### PR TITLE
fix(discord-utilities): update invite link regex

### PR DESCRIPTION
**Pull request by EvolutionX-10**

### CODE REVIEW

**Comment by vladfrangu** [comment]
Please show strings that didn't match before but now do... Or just..what does this exactly change? It looks almost the same to me

**Comment by EvolutionX-10** [comment]
alr @kyranet @vladfrangu


## Previous regex
image
image


## New regex
image
image

**Comment by EvolutionX-10** [comment]
I dont think I'am able to see the image @imranbarbhuiya 🤔

**Comment by imranbarbhuiya** [comment]
> I dont think I'am able to see the image @imranbarbhuiya 🤔

It's matching invalid codes like accepts space

**Comment by EvolutionX-10** [comment]
How about this one @imranbarbhuiya 
`/(https:\/\/)?(www\.)?(((discord(app)?)?\.com\/invite)|((discord(app)?)?\.gg))\/(?<invite>\w+)/i`

File: packages/discord-utilities/src/lib/regexes.ts [hunk 1]
```diff
@@ -29,10 +29,10 @@ export const DiscordHostnameRegex =
 
 /**
  * Regex that can can capture the code of Discord invite links
- * @raw `/^(?:https?:\/\/)?(?:www\.)?(?:discord\.gg\/|discord(?:app)?\.com\/invite\/)?(?<code>[\w\d-]{2,})$/i`
+ * @raw `/(https:\/\/)?(www\.)?(((discord(app)?)?\.com\/invite)|((discord(app)?)?\.gg))\/(?<invite>.+)/i`
```
Comment (imranbarbhuiya) [inline packages/discord-utilities/src/lib/regexes.ts:32]:
```suggestion
 * @raw `/discord(?:(?:app)?\.com\/invite|\.gg(?:\/invite)?)\/([\w-]{2,255})/i`
```

Maybe we can use this one which is used in djs

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

**Comment by imranbarbhuiya** [comment]
> How about this one @imranbarbhuiya `/(https:\/\/)?(www\.)?(((discord(app)?)?\.com\/invite)|((discord(app)?)?\.gg))\/(?<invite>\S+)/i`

You are removing support for 
1. links starting with `http`. 
2. Previously only invite code was present in capturing block. Now u r capturing all the blocks.

**Comment by EvolutionX-10** [comment]
> 1. You are removing support for links starting with `http`.

Nope not really
image

> 2. Previously only invite code was present in capturing block. Now u r capturing all the blocks.

Yeah i didn't notice that before, thanks, let @vladfrangu confirm for the d.js one, once he tells I'll commit the d.js one

**Comment by imranbarbhuiya** [comment]
> > 1. You are removing support for links starting with `http`.
> 
> Nope not really image
> 

Well you missed `^` so it'll match
`adiscord.gg/abc`

**Comment by EvolutionX-10** [comment]
I can imagine the abuse of `abcjoinmyserverdiscord.gg/sapphiredev` but anyways thats user's way to handle it

**Comment by imranbarbhuiya** [comment]
> I can imagine the abuse of `abcjoinmyserverdiscord.gg/sapphiredev` but anyways thats user's way to handle it

still both 1 & 2 aren't resolved

**Comment by EvolutionX-10** [comment]
what is not resolved?

**Comment by imranbarbhuiya** [comment]
1. http
2. capture block

**Comment by EvolutionX-10** [comment]
i don't get the capture group question

**Comment by EvolutionX-10** [comment]
prettier f, gimme a min to fix

**Comment by vladfrangu** [comment]
Is this regex the one in d.js slightly altered? Does it work properly now? Does it support http(s)-less invites? 👀

**Comment by imranbarbhuiya** [comment]
> Is this regex the one in d.js slightly altered?
No
> Does it support http(s)-less invites?
yes

**Comment by EvolutionX-10** [comment]
@vladfrangu now the regex is same as discord.js except there's a named group `code` now that captures the invite link code

File: packages/discord-utilities/src/lib/regexes.ts [hunk 2]
```diff
@@ -29,10 +29,10 @@ export const DiscordHostnameRegex =
 
 /**
  * Regex that can can capture the code of Discord invite links
- * @raw `/^(?:https?:\/\/)?(?:www\.)?(?:discord\.gg\/|discord(?:app)?\.com\/invite\/)?(?<code>[\w\d-]{2,})$/i`
+ * @raw `/discord(?:(?:app)?\.com\/invite|\.gg(?:\/invite)?)\/(?<code>[\w-]{2,255})/gi`
  * @remark Capture group 1 is the invite URL's unique code. It is named `code`.
  */
-export const DiscordInviteLinkRegex = /^(?:https?:\/\/)?(?:www\.)?(?:discord\.gg\/|discord(?:app)?\.com\/invite\/)?(?<code>[\w\d-]{2,})$/i;
+export const DiscordInviteLinkRegex = /discord(?:(?:app)?\.com\/invite|\.gg(?:\/invite)?)\/(?<code>[\w-]{2,255})/gi;
```
Comment (vladfrangu) [inline packages/discord-utilities/src/lib/regexes.ts:35]: We probably want to add `(?:^|\b)` at the start and `(?:$|\b)` at the end

Comment (imranbarbhuiya) [reply packages/discord-utilities/src/lib/regexes.ts:35]: Check my message in discord. You can do like `"discord.gg/invite"`

Comment (EvolutionX-10) [reply packages/discord-utilities/src/lib/regexes.ts:35]: ### Looks like @vladfrangu's way works good @imranbarbhuiya 
image

**Review by vladfrangu — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**